### PR TITLE
Add a rationale to proposals entity

### PR DESCRIPTION
**Pull request by ValentinTrinque**

### CODE REVIEW

**Review by tguinot — APPROVED** [review]

File: generated/json/data-node/api/v1/GetProposalsResponse.json [hunk 1]
```diff
@@ -757,13 +757,37 @@
...
             "additionalProperties": true,
             "type": "object",
             "title": "Proposal",
             "description": "Governance proposal"
         },
+        "vega.ProposalRationale": {
+            "properties": {
+                "description": {
+                    "type": "string",
+                    "description": "Description to show a short title / something in case the link goes offline. This is to be between 0 and 255 unicode characters. This is mandatory for all proposal."
```
[the remark this answers is not in the document]
Comment (ValentinTrinque) [reply generated/json/data-node/api/v1/GetProposalsResponse.json:776]: Nice catch ! Fixed few other things

**Review by pscott31 — APPROVED** [review]

File: generated/json/data-node/api/v1/GetNetworkParametersProposalsResponse.json [hunk 2]
```diff
@@ -757,13 +757,37 @@
                 "error_details": {
                     "type": "string",
                     "description": "The detailed error associated to the reason."
+                },
+                "rational": {
+                    "$ref": "#/definitions/vega.ProposalRationale",
+                    "additionalProperties": true,
+                    "description": "Proposal rational that summarises the change and link to the complete proposed changed."
```
Comment (candida-d) [inline generated/json/data-node/api/v1/GetNetworkParametersProposalsResponse.json:764]:
```suggestion
                    "description": "Proposal rationale that summarises the change and links to the complete proposed changed."
```

File: generated/json/data-node/api/v1/GetNewAssetProposalsResponse.json [hunk 3]
```diff
@@ -757,13 +757,37 @@
                 "error_details": {
                     "type": "string",
                     "description": "The detailed error associated to the reason."
+                },
+                "rational": {
+                    "$ref": "#/definitions/vega.ProposalRationale",
+                    "additionalProperties": true,
+                    "description": "Proposal rational that summarises the change and link to the complete proposed changed."
```
Comment (candida-d) [inline generated/json/data-node/api/v1/GetNewAssetProposalsResponse.json:764]:
```suggestion
                    "description": "Proposal rationale that summarises the change and link to the complete proposed changed."
```

File: generated/json/data-node/api/v1/GetNewFreeformProposalsResponse.json [hunk 4]
```diff
@@ -757,13 +757,37 @@
                 "error_details": {
                     "type": "string",
                     "description": "The detailed error associated to the reason."
+                },
+                "rational": {
+                    "$ref": "#/definitions/vega.ProposalRationale",
+                    "additionalProperties": true,
+                    "description": "Proposal rational that summarises the change and link to the complete proposed changed."
```
Comment (candida-d) [inline generated/json/data-node/api/v1/GetNewFreeformProposalsResponse.json:764]:
```suggestion
                    "description": "Proposal rationale that summarises the change and links to the complete proposed changed."
```

File: generated/json/data-node/api/v1/GetNetworkParametersProposalsResponse.json [hunk 5]
```diff
@@ -757,13 +757,37 @@
                 "error_details": {
                     "type": "string",
                     "description": "The detailed error associated to the reason."
+                },
+                "rational": {
```
Comment (candida-d) [inline generated/json/data-node/api/v1/GetNetworkParametersProposalsResponse.json:761]: @ValentinTrinque Should this be rationale?

File: sources/vega/governance.proto [hunk 6]
```diff
@@ -275,6 +275,8 @@ message Proposal {
   ProposalError reason = 7;
   // The detailed error associated to the reason.
   string error_details = 8;
+  // Proposal rationale that summarises the change and links to the complete proposed changed.
+  ProposalRationale rational = 9;
```
Comment (wwestgarth) [inline sources/vega/governance.proto:279]: For consistency shouldn't this field also be called `rationale`?

File: generated/json/data-node/api/v1/GetNewAssetProposalsResponse.json [hunk 7]
```diff
@@ -757,13 +757,37 @@
                 "error_details": {
                     "type": "string",
                     "description": "The detailed error associated to the reason."
+                },
+                "rational": {
```
Comment (candida-d) [inline generated/json/data-node/api/v1/GetNewAssetProposalsResponse.json:761]: Same question here - rationale?

File: generated/json/data-node/api/v1/GetNewFreeformProposalsResponse.json [hunk 8]
```diff
@@ -757,13 +757,37 @@
                 "error_details": {
                     "type": "string",
                     "description": "The detailed error associated to the reason."
+                },
+                "rational": {
```
Comment (candida-d) [inline generated/json/data-node/api/v1/GetNewFreeformProposalsResponse.json:761]: And here :)

File: generated/json/data-node/api/v1/GetNewMarketProposalsResponse.json [hunk 9]
```diff
@@ -757,13 +757,37 @@
                 "error_details": {
                     "type": "string",
                     "description": "The detailed error associated to the reason."
+                },
+                "rational": {
```
Comment (candida-d) [inline generated/json/data-node/api/v1/GetNewMarketProposalsResponse.json:761]: rationale?

**Review by candida-d — APPROVED** [review]

**Pull request merged**